js/components/controlBar.js

```diff
@@ -150,12 +163,46 @@ var ControlBar = React.createClass({
     }
   },
 
+  /**
+   * Fires when a key is pressed on the video quality button.
+   * @private
+   * @param {type} event The keydown event object.
+   */
+  handleQualityKeyDown: function(event) {
+    switch (event.key) {
+      case CONSTANTS.KEY_VALUES.SPACE:
+      case CONSTANTS.KEY_VALUES.ENTER:
+      // Ctrl and Alt are needed as a workaround for VoiceOver, which uses the
+      // CTRL + OPTION + SPACE combination to activate buttons. VoiceOver actually
+      // suppresses the spacebar keyboard event when this combination is used, so we
+      // can only detect either CTRL or OPTION. This can obviously fail if the user
+      // presses a different key after CTRL + OPTION, but a false positive is preferred.
+      case CONSTANTS.KEY_VALUES.CONTROL:
+      case CONSTANTS.KEY_VALUES.ALT:
+        this.qualityMenuOpenedWithKeyboard = true;
+        break;
+      default:
+        break;
+    }
+  },
+
   toggleQualityPopover: function () {
     this.props.controller.toggleVideoQualityPopOver();
   },
 
-  closeQualityPopover: function () {
-    if (this.props.controller.state.videoQualityOptions.showVideoQualityPopover == true) {
+  closeQualityPopover: function (params) {
+    params = params || {};
+
+    if (this.props.controller.state.videoQualityOptions.showVideoQualityPopover === true) {
+      // Re-focus on quality button when closing the quality popover if the latter was
+      // originally opened with a key press.
+      if (params.restoreToggleButtonFocus &&
+          this.qualityMenuOpenedWithKeyboard &&
+          this.qualityBtnElement &&
+          typeof this.qualityBtnElement.focus === 'function') {
+        this.qualityBtnElement.focus();
+      }
+      this.qualityMenuOpenedWithKeyboard = false;
```

Comment:
What do you think of storing the focused element of the control bar when entering a submenu (like the quality options)? And we when close a submenu, restore the focused element, regardless of whether keyboard or mouse were used.